### PR TITLE
functionals max_pool2d, batch_norm, relu, interpolate, cross_entropy, dropout

### PR DESCRIPTION
**Pull request by izdeby**
Enabled several nn functions to work with nested tensors.
Covered: 
- max_pool2d
- batch_norm
- relu
- interpolate
- cross_entropy
- dropout

Tested via unit tests

Todo: 
1. move the implementations to C++ for performance improvements.

### CODE REVIEW

File: nestedtensor/nested/functions.py [hunk 1]
```diff
@@ -141,29 +133,11 @@ def _embedding_bag(input, *args, **kwargs):
...
-            input, running_mean, running_var, weight, bias, training, momentum, eps)
-        if squeeze_after:
-            result = result.squeeze(0)
-        return result
-
-    tw_batch_norm = utils.tensorwise()(t_batch_norm)
-
-    return tw_batch_norm(input, running_mean, running_var, weight, bias, training, momentum, eps)
-
+    validate_nt(input)
```
Comment (cpuhrsch) [inline nestedtensor/nested/functions.py]: You also want to make sure the other arguments are either Tensors or NestedTensors as appropiate

File: nestedtensor/nested/functions.py [hunk 2]
```diff
@@ -141,29 +133,11 @@ def _embedding_bag(input, *args, **kwargs):
...
+    tensor, mask = input.to_tensor_mask()
+    res = orig_batch_norm(tensor, running_mean, running_var, weight, bias, training, momentum, eps)
```
Comment (cpuhrsch) [inline nestedtensor/nested/functions.py]: You can pass in the "torch.batch_norm" function as an argument within ```__torch_function__```.

File: nestedtensor/nested/functions.py [hunk 3]
```diff
@@ -147,42 +141,93 @@ def _embedding_bag(input, *args, **kwargs):
                                               per_sample_weights)
 
 
-orig_batch_norm = torch.nn.functional.batch_norm
+def batch_norm(input, running_mean, running_var, weight=None, bias=None, training=False, momentum=0.1, eps=1e-05):
+    validate_nt(input)
+    if running_mean is not None and not isinstance(running_mean, torch.Tensor):
```
Comment (cpuhrsch) [inline nestedtensor/nested/functions.py]: Use torch.is_tensor

**Comment by izdeby** [comment]
@cpuhrsch, as discussed offline, since last review: 
1) updated tests to use both NT constructors
2) updated interpolate test to use different size types (None, tuple, int) 

Autograd testing will come as a follow up pr due to a bug with as_nested_tensor.

**Comment by izdeby** [comment]
@cpuhrsch, should be ready. Grad checks were added to the integration test

File: nestedtensor/nested/monkey_patch.py [hunk 4]
```diff
@@ -233,9 +233,14 @@ def new_fn(self):
     #     set_nt_method(function_name, getattr(functions, function_name))
 
     set_function(torch.conv2d, functions.conv2d)
-    set_function(torch.max_pool2d, functions.max_pool2d)
+    set_function(torch.nn.functional.max_pool2d, functions.max_pool2d)
```
Comment (cpuhrsch) [inline nestedtensor/nested/monkey_patch.py]: You should be overriding the "innermost" function and not the functional. So, "torch.max_pool2d" and not "torch.nn.functional.max_pool2d". I'm actually surprised this works via ```__torch_function__```.

Comment (cpuhrsch) [reply nestedtensor/nested/monkey_patch.py]: I see, it's due to the [torch function support for functionals](https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L2979). Eventually however we'll want to overwrite builtins to implement NestedTensor C++ kernels for natives functions and get JIT support.
